BUG: Runtime panic when deploying contracts

### Problem

Got a runtime panic when trying to deploy contracts as part of automated tests

### Steps to Reproduce

1. Use this branch of the flow-ft repo: https://github.com/onflow/flow-ft/tree/josh/transactions

2. This uses github.com/onflow/flow-go-sdk@v0.7.0
github.com/dapperlabs/flow-emulator@v0.5.0
github.com/dapperlabs/flow-go@30b3e2d5a5860c437e523921d08a97896b1e50e0
github.com/onflow/cadence@v0.5.0

3. Go to `lib/go/test/` and run `go test -v`

4. See the runtime panic

## Full Stack Trace

```
=== RUN   TestTokenDeployment
--- FAIL: TestTokenDeployment (0.33s)
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x70 pc=0x141a25c]

goroutine 36 [running]:
testing.tRunner.func1.1(0x1936360, 0x24f2d00)
	/usr/local/Cellar/go/1.14/libexec/src/testing/testing.go:941 +0x3d0
testing.tRunner.func1(0xc0001ccc60)
	/usr/local/Cellar/go/1.14/libexec/src/testing/testing.go:944 +0x3f9
panic(0x1936360, 0x24f2d00)
	/usr/local/Cellar/go/1.14/libexec/src/runtime/panic.go:967 +0x15d
github.com/onflow/cadence/runtime/interpreter.recoverErrors(0xc000674710)
	/Users/joshuahannan/go/pkg/mod/github.com/onflow/cadence@v0.5.0/runtime/interpreter/interpreter.go:864 +0x15a
panic(0x1936360, 0x24f2d00)
	/usr/local/Cellar/go/1.14/libexec/src/runtime/panic.go:967 +0x15d
github.com/onflow/cadence/runtime/sema.(*CapabilityType).Unify(0xc0001930a0, 0x1d12660, 0xc0001930a0, 0xc000910a50, 0xc000541b80, 0x922, 0x42, 0x2d, 0x92a, 0x42, ...)
	/Users/joshuahannan/go/pkg/mod/github.com/onflow/cadence@v0.5.0/runtime/sema/type.go:6652 +0x4c
github.com/onflow/cadence/runtime/sema.(*Checker).checkInvocationRequiredArgument(0xc0000d6300, 0xc0000ac2a0, 0x1, 0x1, 0x0, 0xc00057b540, 0xc000541b20, 0x1, 0x1, 0xc000910a50, ...)
	/Users/joshuahannan/go/pkg/mod/github.com/onflow/cadence@v0.5.0/runtime/sema/check_invocation_expression.go:481 +0x28a
github.com/onflow/cadence/runtime/sema.(*Checker).checkInvocation(0xc0000d6300, 0xc0001e24e0, 0x1d15a00, 0xc000541a50, 0xc000541a50, 0x1, 0x0, 0x0, 0x0)
	/Users/joshuahannan/go/pkg/mod/github.com/onflow/cadence@v0.5.0/runtime/sema/check_invocation_expression.go:380 +0x27f
github.com/onflow/cadence/runtime/sema.(*Checker).checkInvocationExpression.func3(...)
	/Users/joshuahannan/go/pkg/mod/github.com/onflow/cadence@v0.5.0/runtime/sema/check_invocation_expression.go:97
github.com/onflow/cadence/runtime/sema.(*Checker).checkInvocationExpression(0xc0000d6300, 0xc0001e24e0, 0x0, 0x0)
	/Users/joshuahannan/go/pkg/mod/github.com/onflow/cadence@v0.5.0/runtime/sema/check_invocation_expression.go:107 +0x7af
github.com/onflow/cadence/runtime/sema.(*Checker).VisitInvocationExpression(0xc0000d6300, 0xc0001e24e0, 0xc0000d6300, 0x2f153e8)
	/Users/joshuahannan/go/pkg/mod/github.com/onflow/cadence@v0.5.0/runtime/sema/check_invocation_expression.go:27 +0x3f
github.com/onflow/cadence/runtime/ast.(*InvocationExpression).AcceptExp(...)
	/Users/joshuahannan/go/pkg/mod/github.com/onflow/cadence@v0.5.0/runtime/ast/expression.go:339
github.com/onflow/cadence/runtime/ast.(*InvocationExpression).Accept(...)
	/Users/joshuahannan/go/pkg/mod/github.com/onflow/cadence@v0.5.0/runtime/ast/expression.go:335
github.com/onflow/cadence/runtime/sema.(*Checker).VisitCreateExpression(0xc0000d6300, 0xc0002a1280, 0x0, 0x0)
	/Users/joshuahannan/go/pkg/mod/github.com/onflow/cadence@v0.5.0/runtime/sema/check_create_expression.go:37 +0xd8
github.com/onflow/cadence/runtime/ast.(*CreateExpression).AcceptExp(...)
	/Users/joshuahannan/go/pkg/mod/github.com/onflow/cadence@v0.5.0/runtime/ast/expression.go:646
github.com/onflow/cadence/runtime/ast.(*CreateExpression).Accept(0xc0002a1280, 0x1d2e1c0, 0xc0000d6300, 0xc000671458, 0x142aaf9)
	/Users/joshuahannan/go/pkg/mod/github.com/onflow/cadence@v0.5.0/runtime/ast/expression.go:642 +0x5f
github.com/onflow/cadence/runtime/sema.(*Checker).VisitUnaryExpression(0xc0000d6300, 0xc0003a6870, 0xc0000d6300, 0x2f153e8)
	/Users/joshuahannan/go/pkg/mod/github.com/onflow/cadence@v0.5.0/runtime/sema/check_unary_expression.go:28 +0x62
github.com/onflow/cadence/runtime/ast.(*UnaryExpression).AcceptExp(...)
	/Users/joshuahannan/go/pkg/mod/github.com/onflow/cadence@v0.5.0/runtime/ast/expression.go:507
github.com/onflow/cadence/runtime/ast.(*UnaryExpression).Accept(0xc0003a6870, 0x1d2e1c0, 0xc0000d6300, 0xc000671560, 0x100c1ea)
	/Users/joshuahannan/go/pkg/mod/github.com/onflow/cadence@v0.5.0/runtime/ast/expression.go:503 +0x62
github.com/onflow/cadence/runtime/sema.(*Checker).VisitReturnStatement(0xc0000d6300, 0xc0002d7d00, 0x0, 0x0)
	/Users/joshuahannan/go/pkg/mod/github.com/onflow/cadence@v0.5.0/runtime/sema/check_return_statement.go:56 +0x292
github.com/onflow/cadence/runtime/ast.(*ReturnStatement).Accept(0xc0002d7d00, 0x1d2e1c0, 0xc0000d6300, 0x1ced901, 0x1ced9a0)
	/Users/joshuahannan/go/pkg/mod/github.com/onflow/cadence@v0.5.0/runtime/ast/statement.go:36 +0x3e
github.com/onflow/cadence/runtime/sema.(*Checker).visitStatements(0xc0000d6300, 0xc0003142d0, 0x1, 0x1)
	/Users/joshuahannan/go/pkg/mod/github.com/onflow/cadence@v0.5.0/runtime/sema/check_block.go:68 +0x14b
github.com/onflow/cadence/runtime/sema.(*Checker).visitFunctionBlock.func1()
	/Users/joshuahannan/go/pkg/mod/github.com/onflow/cadence@v0.5.0/runtime/sema/check_function.go:350 +0x4b
github.com/onflow/cadence/runtime/sema.(*Checker).visitWithPostConditions(0xc0000d6300, 0x0, 0x1d12760, 0xc0007d72c0, 0xc0006717d8)
	/Users/joshuahannan/go/pkg/mod/github.com/onflow/cadence@v0.5.0/runtime/sema/check_function.go:308 +0x54
github.com/onflow/cadence/runtime/sema.(*Checker).visitFunctionBlock(0xc0000d6300, 0xc0002a12c0, 0xc0001729e0, 0x1)
	/Users/joshuahannan/go/pkg/mod/github.com/onflow/cadence@v0.5.0/runtime/sema/check_function.go:343 +0xcf
github.com/onflow/cadence/runtime/sema.(*Checker).checkFunction.func2()
	/Users/joshuahannan/go/pkg/mod/github.com/onflow/cadence@v0.5.0/runtime/sema/check_function.go:158 +0x16f
github.com/onflow/cadence/runtime/sema.(*FunctionActivations).WithFunction(0xc0002a1560, 0xc00057b640, 0x3, 0xc000671928)
	/Users/joshuahannan/go/pkg/mod/github.com/onflow/cadence@v0.5.0/runtime/sema/function_activations.go:67 +0xfa
github.com/onflow/cadence/runtime/sema.(*Checker).checkFunction(0xc0000d6300, 0xc0004138b0, 0xc0003a6660, 0xc00057b640, 0xc0002a12c0, 0x1, 0x0, 0x1)
	/Users/joshuahannan/go/pkg/mod/github.com/onflow/cadence@v0.5.0/runtime/sema/check_function.go:141 +0x17a
github.com/onflow/cadence/runtime/sema.(*Checker).visitFunctionDeclaration(0xc0000d6300, 0xc0001e2540, 0xc000010001, 0x19559c0, 0xc000910900)
	/Users/joshuahannan/go/pkg/mod/github.com/onflow/cadence@v0.5.0/runtime/sema/check_function.go:77 +0x13d
github.com/onflow/cadence/runtime/sema.(*Checker).checkCompositeFunctions.func1(0xc0000d6300, 0xc0007d7180, 0xc000671a90)
	/Users/joshuahannan/go/pkg/mod/github.com/onflow/cadence@v0.5.0/runtime/sema/check_composite_declaration.go:1386 +0xa2
github.com/onflow/cadence/runtime/sema.(*Checker).checkCompositeFunctions(0xc0000d6300, 0xc0000ac2b0, 0x1, 0x1, 0xc0007d7180)
	/Users/joshuahannan/go/pkg/mod/github.com/onflow/cadence@v0.5.0/runtime/sema/check_composite_declaration.go:1394 +0x7d
github.com/onflow/cadence/runtime/sema.(*Checker).visitCompositeDeclaration(0xc0000d6300, 0xc000222780, 0x2)
	/Users/joshuahannan/go/pkg/mod/github.com/onflow/cadence@v0.5.0/runtime/sema/check_composite_declaration.go:115 +0x5ca
github.com/onflow/cadence/runtime/sema.(*Checker).VisitCompositeDeclaration(0xc0000d6300, 0xc000222780, 0x0, 0x0)
	/Users/joshuahannan/go/pkg/mod/github.com/onflow/cadence@v0.5.0/runtime/sema/check_composite_declaration.go:28 +0x3e
github.com/onflow/cadence/runtime/ast.(*CompositeDeclaration).Accept(0xc000222780, 0x1d2e1c0, 0xc0000d6300, 0x2, 0x1054d3c)
	/Users/joshuahannan/go/pkg/mod/github.com/onflow/cadence@v0.5.0/runtime/ast/composite.go:41 +0x3b
github.com/onflow/cadence/runtime/sema.(*Checker).VisitProgram(0xc0000d6300, 0xc0001b8f20, 0x10896c6, 0x5f063b6d)
	/Users/joshuahannan/go/pkg/mod/github.com/onflow/cadence@v0.5.0/runtime/sema/checker.go:453 +0x417
github.com/onflow/cadence/runtime/ast.(*Program).Accept(...)
	/Users/joshuahannan/go/pkg/mod/github.com/onflow/cadence@v0.5.0/runtime/ast/program.go:53
github.com/onflow/cadence/runtime/sema.(*Checker).Check.func1()
	/Users/joshuahannan/go/pkg/mod/github.com/onflow/cadence@v0.5.0/runtime/sema/checker.go:326 +0x3b
github.com/onflow/cadence/runtime.(*interpreterRuntime).parseAndCheckProgram.func3.1()
	/Users/joshuahannan/go/pkg/mod/github.com/onflow/cadence@v0.5.0/runtime/runtime.go:524 +0x24
github.com/onflow/cadence/runtime.reportMetric(0xc000673b00, 0x1d1c520, 0xc0000d6800, 0xc000673b10)
	/Users/joshuahannan/go/pkg/mod/github.com/onflow/cadence@v0.5.0/runtime/runtime.go:104 +0x93
github.com/onflow/cadence/runtime.(*interpreterRuntime).parseAndCheckProgram.func3(0x1ced640, 0xc0002e1ce0, 0xc000315050)
	/Users/joshuahannan/go/pkg/mod/github.com/onflow/cadence@v0.5.0/runtime/runtime.go:522 +0x97
github.com/onflow/cadence/runtime/sema.(*Checker).Check(0xc0000d6300, 0x1ced640, 0xc0002e1ce0)
	/Users/joshuahannan/go/pkg/mod/github.com/onflow/cadence@v0.5.0/runtime/sema/checker.go:329 +0x128
github.com/onflow/cadence/runtime.(*interpreterRuntime).parseAndCheckProgram(0xc0002499c0, 0xc0003d2a80, 0x935, 0x935, 0x1d1c520, 0xc0000d6800, 0x1ced640, 0xc0002e1ce0, 0xc000146a80, 0x8, ...)
	/Users/joshuahannan/go/pkg/mod/github.com/onflow/cadence@v0.5.0/runtime/runtime.go:540 +0x42d
github.com/onflow/cadence/runtime.(*interpreterRuntime).updateAccountCode(0xc0002499c0, 0x1d1c520, 0xc0000d6800, 0xc0004e4720, 0xc0003d2a80, 0x935, 0x935, 0xee5e8fa7c1d2fcf3, 0xc0002c0b90, 0x0, ...)
	/Users/joshuahannan/go/pkg/mod/github.com/onflow/cadence@v0.5.0/runtime/runtime.go:1080 +0x181
github.com/onflow/cadence/runtime.(*interpreterRuntime).newSetCodeFunction.func1(0x0, 0xc0002c0b90, 0x1, 0x1, 0xc00007c0c0, 0x1, 0x1, 0xc000410330, 0x1ced620, 0xc0004e4700, ...)
	/Users/joshuahannan/go/pkg/mod/github.com/onflow/cadence@v0.5.0/runtime/runtime.go:1032 +0x1fd
github.com/onflow/cadence/runtime/interpreter.HostFunctionValue.Invoke(...)
	/Users/joshuahannan/go/pkg/mod/github.com/onflow/cadence@v0.5.0/runtime/interpreter/function.go:155
github.com/onflow/cadence/runtime/interpreter.(*Interpreter).functionValueInvocationTrampoline(0xc0000b5800, 0x1d0cf60, 0xc0002c0400, 0xc0002c0b40, 0x1, 0x1, 0xc00007c0c0, 0x1, 0x1, 0xc00007c0d0, ...)
	/Users/joshuahannan/go/pkg/mod/github.com/onflow/cadence@v0.5.0/runtime/interpreter/interpreter.go:2098 +0x311
github.com/onflow/cadence/runtime/interpreter.(*Interpreter).VisitInvocationExpression.func1.1(0x1a1af60, 0xc0000de6c0, 0xc0002e16c0, 0x1339a72401cfeaa0)
	/Users/joshuahannan/go/pkg/mod/github.com/onflow/cadence@v0.5.0/runtime/interpreter/interpreter.go:2016 +0x2c9
github.com/onflow/cadence/runtime/trampoline.FlatMap.FlatMap.func1(0x1a1af60, 0xc0000de6c0, 0xc0000de6c0, 0xc0002e1c20)
	/Users/joshuahannan/go/pkg/mod/github.com/onflow/cadence@v0.5.0/runtime/trampoline/trampoline.go:142 +0x42
github.com/onflow/cadence/runtime/trampoline.FlatMap.FlatMap.func1(0x1a1af60, 0xc0000de6c0, 0x18df080, 0xc0002e1c40)
	/Users/joshuahannan/go/pkg/mod/github.com/onflow/cadence@v0.5.0/runtime/trampoline/trampoline.go:142 +0x42
github.com/onflow/cadence/runtime/trampoline.FlatMap.Resume.func1(0xc0002e1c20, 0x18df080)
	/Users/joshuahannan/go/pkg/mod/github.com/onflow/cadence@v0.5.0/runtime/trampoline/trampoline.go:154 +0x37
github.com/onflow/cadence/runtime/interpreter.(*Interpreter).runUntilNextStatement(0xc0000b5800, 0x1cfeaa0, 0xc0002e1c20, 0x0, 0x0, 0xc0002e1720)
	/Users/joshuahannan/go/pkg/mod/github.com/onflow/cadence@v0.5.0/runtime/interpreter/interpreter.go:608 +0x35
github.com/onflow/cadence/runtime/interpreter.(*Interpreter).runAllStatements(0xc0000b5800, 0x1cfeaa0, 0xc00000e160, 0x3, 0x4)
	/Users/joshuahannan/go/pkg/mod/github.com/onflow/cadence@v0.5.0/runtime/interpreter/interpreter.go:620 +0x56
github.com/onflow/cadence/runtime/interpreter.(*Interpreter).InvokeTransaction(0xc0000b5800, 0x0, 0xc00004cec0, 0x3, 0x4, 0x0, 0x0)
	/Users/joshuahannan/go/pkg/mod/github.com/onflow/cadence@v0.5.0/runtime/interpreter/interpreter.go:853 +0x118
github.com/onflow/cadence/runtime.(*interpreterRuntime).transactionExecutionFunction.func1(0xc0000b5800, 0x0, 0x0, 0x2854a17800472850, 0x5f063b6d)
	/Users/joshuahannan/go/pkg/mod/github.com/onflow/cadence@v0.5.0/runtime/runtime.go:400 +0x183
github.com/onflow/cadence/runtime.(*interpreterRuntime).interpret.func1()
	/Users/joshuahannan/go/pkg/mod/github.com/onflow/cadence@v0.5.0/runtime/runtime.go:232 +0xa4
github.com/onflow/cadence/runtime.reportMetric(0xc000472978, 0x1d1c520, 0xc0000d6800, 0xc000674960)
	/Users/joshuahannan/go/pkg/mod/github.com/onflow/cadence@v0.5.0/runtime/runtime.go:104 +0x93
github.com/onflow/cadence/runtime.(*interpreterRuntime).interpret(0xc0002499c0, 0x1ced620, 0xc0004e4700, 0x1d1c520, 0xc0000d6800, 0xc0004e4720, 0xc0000d6900, 0xc000146380, 0x8, 0xc, ...)
	/Users/joshuahannan/go/pkg/mod/github.com/onflow/cadence@v0.5.0/runtime/runtime.go:226 +0x1d2
github.com/onflow/cadence/runtime.(*interpreterRuntime).ExecuteTransaction(0xc0002499c0, 0xc000654340, 0xce, 0xce, 0xc000715c80, 0x2, 0x2, 0x1d1c520, 0xc0000d6800, 0x1ced620, ...)
	/Users/joshuahannan/go/pkg/mod/github.com/onflow/cadence@v0.5.0/runtime/runtime.go:339 +0x57d
github.com/dapperlabs/flow-go/fvm.(*TransactionInvocator).Process(0x2536258, 0xc000520b30, 0x1d08d40, 0x24f51c0, 0x1cf36e0, 0xc0005241f0, 0x1ced4c0, 0xc0001ccd80, 0x0, 0x186a0, ...)
	/Users/joshuahannan/go/pkg/mod/github.com/dapperlabs/flow-go@v0.4.1-0.20200708192840-30b3e2d5a586/fvm/transaction.go:90 +0x272
github.com/dapperlabs/flow-go/fvm.(*TransactionProcedure).Run(0xc000356cb0, 0xc000520b30, 0x1d08d40, 0x24f51c0, 0x1cf36e0, 0xc0005241f0, 0x1ced4c0, 0xc0001ccd80, 0x0, 0x186a0, ...)
	/Users/joshuahannan/go/pkg/mod/github.com/dapperlabs/flow-go@v0.4.1-0.20200708192840-30b3e2d5a586/fvm/transaction.go:37 +0xe4
github.com/dapperlabs/flow-go/fvm.(*VirtualMachine).Run(...)
	/Users/joshuahannan/go/pkg/mod/github.com/dapperlabs/flow-go@v0.4.1-0.20200708192840-30b3e2d5a586/fvm/fvm.go:29
github.com/dapperlabs/flow-emulator.(*Blockchain).executeNextTransaction.func1(0xc00057ae40, 0xc0000b5a40, 0xc000475230, 0xc0000d8ec8, 0x1000000016681e3)
	/Users/joshuahannan/go/pkg/mod/github.com/dapperlabs/flow-emulator@v0.5.0/blockchain.go:613 +0x1c7
github.com/dapperlabs/flow-emulator.(*pendingBlock).ExecuteNextTransaction(0xc0001110e0, 0xc0004754a8, 0x1ced4c0, 0xc0001ccd80, 0x0)
	/Users/joshuahannan/go/pkg/mod/github.com/dapperlabs/flow-emulator@v0.5.0/pendingBlock.go:148 +0x10f
github.com/dapperlabs/flow-emulator.(*Blockchain).executeNextTransaction(0xc0001ccd80, 0x1d08d40, 0x24f51c0, 0x1cf36e0, 0xc0005241f0, 0x1ced4c0, 0xc0001ccd80, 0x0, 0x186a0, 0xc000210620, ...)
	/Users/joshuahannan/go/pkg/mod/github.com/dapperlabs/flow-emulator@v0.5.0/blockchain.go:606 +0xa7
github.com/dapperlabs/flow-emulator.(*Blockchain).executeBlock(0xc0001ccd80, 0xce, 0xc000715c80, 0x2, 0x2, 0x3b9ac9ff)
	/Users/joshuahannan/go/pkg/mod/github.com/dapperlabs/flow-emulator@v0.5.0/blockchain.go:570 +0x20e
github.com/dapperlabs/flow-emulator.(*Blockchain).executeAndCommitBlock(0xc0001ccd80, 0xc000654340, 0xce, 0xce, 0xc000715c80, 0x2, 0x2)
	/Users/joshuahannan/go/pkg/mod/github.com/dapperlabs/flow-emulator@v0.5.0/blockchain.go:694 +0x43
github.com/dapperlabs/flow-emulator.(*Blockchain).CreateAccount(0xc0001ccd80, 0xc000475f58, 0x1, 0x1, 0xc0006bca80, 0x935, 0xa80, 0x0, 0x0, 0x0)
	/Users/joshuahannan/go/pkg/mod/github.com/dapperlabs/flow-emulator@v0.5.0/blockchain.go:812 +0x393
github.com/onflow/flow-ft/lib/go/test.DeployTokenContracts(0xc0001ccd80, 0xc0001ccc60, 0xc000475f58, 0x1, 0x1, 0x0, 0x0, 0x0)
	/Users/joshuahannan/flow/flow-ft/lib/go/test/token_test.go:433 +0x29e
github.com/onflow/flow-ft/lib/go/test.TestTokenDeployment(0xc0001ccc60)
	/Users/joshuahannan/flow/flow-ft/lib/go/test/token_test.go:24 +0xeb
testing.tRunner(0xc0001ccc60, 0x1bd0c80)
	/usr/local/Cellar/go/1.14/libexec/src/testing/testing.go:992 +0xdc
created by testing.(*T).Run
	/usr/local/Cellar/go/1.14/libexec/src/testing/testing.go:1043 +0x357
exit status 2
```

--- a/runtime/sema/type.go
+++ b/runtime/sema/type.go
@@ -6644,18 +6644,21 @@ func (t *CapabilityType) Unify(
 	report func(err error),
 	outerRange ast.Range,
 ) bool {
-	otherOptional, ok := other.(*CapabilityType)
+	otherCap, ok := other.(*CapabilityType)
 	if !ok {
 		return false
 	}
 
-	return t.BorrowType.Unify(otherOptional.BorrowType, typeParameters, report, outerRange)
+	if t.BorrowType == nil {
+		return false
+	}
+
+	return t.BorrowType.Unify(otherCap.BorrowType, typeParameters, report, outerRange)
 }
 
 func (t *CapabilityType) Resolve(typeParameters map[*TypeParameter]Type) Type {
 	var resolvedBorrowType Type
 	if t.BorrowType != nil {
-
 		resolvedBorrowType = t.BorrowType.Resolve(typeParameters)
 	}
 
@@ -6694,7 +6697,9 @@ func (t *CapabilityType) BaseType() Type {
 func (t *CapabilityType) TypeArguments() []Type {
 	borrowType := t.BorrowType
 	if borrowType == nil {
-		borrowType = &AnyType{}
+		borrowType = &ReferenceType{
+			Type: &AnyType{},
+		}
 	}
 	return []Type{
 		borrowType,

--- a/runtime/tests/checker/genericfunction_test.go
+++ b/runtime/tests/checker/genericfunction_test.go
@@ -886,3 +886,16 @@ func TestCheckGenericFunctionIsInvalid(t *testing.T) {
 
 	assert.False(t, genericFunctionType.IsInvalidType())
 }
+
+// https://github.com/onflow/cadence/issues/225
+func TestCheckBorrowOfCapabilityWithoutTypeArgument(t *testing.T) {
+
+	t.Parallel()
+
+	_, err := ParseAndCheckWithPanic(t, `
+      let cap: Capability = panic("")
+      let ref = cap.borrow<&Int>()!
+    `)
+
+	require.NoError(t, err)
+}

